### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by Joannis**
A further continuation of performance improvements in #1186 

Touches performance improvements in:

- TCP
- MySQL
- Async

### CODE REVIEW

File: Sources/MySQL/Results/ResultsStream.swift [hunk 1]
```diff
@@ -99,12 +99,12 @@ extension ResultsStream {
         let parser = Parser(packet: packet)
         
         do {
-            let catalog = try parser.parseLenEncString()
-            let database = try parser.parseLenEncString()
-            let table = try parser.parseLenEncString()
-            let originalTable = try parser.parseLenEncString()
+            try parser.skipLenEnc() // let catalog = try parser.parseLenEncString()
```
Comment (tanner0101) [inline Sources/MySQL/Results/ResultsStream.swift]: is this just temporary or do we not actually need that info?

File: Sources/Routing/TrieRouter.swift [hunk 2]
```diff
@@ -57,39 +57,74 @@ public final class TrieRouter: Router {
...
-        if request.method == .options, let methodName = request.headers[.accessControlAllowMethods] {
-             path = [methodName]
-        } else {
-            path = [request.method.string]
-        }
+        // Skip past the first `/`
+        var baseIndex = request.uri.path.index(after: request.uri.path.startIndex)
         
-        path += request.uri.path.split(separator: "/").map(String.init)
+        if baseIndex < request.uri.path.endIndex {
```
Comment (tanner0101) [inline Sources/Routing/TrieRouter.swift]: anyway we could separate this out into a func?

File: Sources/Routing/TrieRouter.swift [hunk 3]
```diff
@@ -57,39 +57,74 @@ public final class TrieRouter: Router {
...
         
         // always start at the root node
         var current: TrieRouterNode = root
-
-        // traverse the constant path supplied
-        var iterator = path.makeIterator()
-        while let path = iterator.next() {
-            if let node = current.findConstantNode(at: path) {
+        
+        func walkingPathSuccessfully<S: StringProtocol>(component: S) -> Bool {
```
Comment (tanner0101) [inline Sources/Routing/TrieRouter.swift]: a private func would be more readable here than a nested one (if possible)

File: Sources/MySQL/Results/ResultsStream.swift (hunk 1 above)
Comment (Joannis) [reply Sources/MySQL/Results/ResultsStream.swift]: We almost never need this info, I'd like to parse this lazily in the future. But this information is currently not used anywhere in MySQL. If you need this in Fluent we can start parsing this (lazily).

Comment (tanner0101) [reply Sources/MySQL/Results/ResultsStream.swift]: Ah, makes sense. Parsing it lazily would be cool but I think we can skip for now.

**Pull request merged**